server/src/main/java/org/schemarepo/server/JsonRenderer.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package org.schemarepo.server;
+
+import java.util.Properties;
+
+import javax.ws.rs.core.MediaType;
+
+import org.schemarepo.RepositoryUtil;
+import org.schemarepo.SchemaEntry;
+import org.schemarepo.Subject;
+import org.schemarepo.json.JsonUtil;
+
+/**
+ * Renders as JSON. The actual work is delegated to utility class {@link org.schemarepo.RepositoryUtil}
+ */
+public class JsonRenderer implements Renderer {
+
+  private final JsonUtil jsonUtil;
+
+  public JsonRenderer(JsonUtil jsonUtil) {
+    if (jsonUtil == null) {
+      throw new IllegalArgumentException("jsonUtil required");
+    }
+    this.jsonUtil = jsonUtil;
+  }
+
+  @Override
+  public String getMediaType() {
+    return MediaType.APPLICATION_JSON;
+  }
+
+  @Override
+  public String renderSubjects(Iterable<Subject> subjects) {
+    return jsonUtil.subjectsToJson(subjects);
+  }
+
+  @Override
+  public String renderSchemas(Iterable<SchemaEntry> schemaEntries) {
+    return jsonUtil.schemasToJson(schemaEntries);
+  }
+
+  @Override
+  public String renderSchemaEntry(SchemaEntry schemaEntry, boolean requestForLatest) {
+    return requestForLatest ? schemaEntry.toString() : schemaEntry.getSchema();
+  }
+
+  @Override
+  public String renderProperties(Properties props, String comment) {
+    // TODO: when the need arises, this may yield an actual JSON
```

Comment:
My previous comment was a bit misleading. I *tested* 4932f5494d29d461cab746b4667b728f06cccb5e, because it is the commit prior to 4bb15507073139ed00ed8c24946c94754ae243fe where the changes to ```Accept``` handling actually happened.

So I checked out 4932f5494d29d461cab746b4667b728f06cccb5e, ran it and curled it with ```Accept: text/html``` and I did get a ```406```. I can post the bash commands to reproduce, if you want.

I think the reason why the Jersey REST client worked without explicitly specifying any Accept header is because its default Accept clause includes ```*/*``` (albeit at a lower priority than ```text/html```, which is what is screwing us up at the moment...). Previously, it was the REST *server* that was annotated with ```@Produces(TEXT_PLAIN)```, not the REST client.

I'm sorry for my confusion, but I'm not sure what exactly do you propose to be the plan?

I'm still under the impression that the only sane way we can deal with this is to have ```/schema-repo``` serving machine-readable (and backwards-compatible) content while ```/schema-repo-browser``` serves the human-readable content...